pages/index.tsx

```diff
@@ -0,0 +1,245 @@
+'use client';
+import { useEffect, useState } from 'react'
+import axios from "axios";
+import Prefectures from "@/utils/Pref";
+import { Select } from 'antd';
+import { MdOutlineLocationOn, MdWbSunny } from "react-icons/md";
+import { MdMyLocation } from "react-icons/md";
+import { WiHumidity,WiStrongWind } from "react-icons/wi";
+import  ForecastWeatherDetail  from "@/components/ForecastWeatherDetails";
+import { format, parseISO } from "date-fns";
+import { LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip,Legend } from 'recharts';
+import { useSearchParams, usePathname, useRouter } from 'next/navigation';
+
+// Format in which current weather api returns data for given location
+interface WeatherDetail {
+  dt: number;
+  dt_txt:string;
+  main: {
+    temp: number;
+    feels_like: number;
+    temp_min: number;
+    temp_max: number;
+    humidity: number;
+  };
+  weather: {
+    id: number;
+    main: string;
+    description: string;
+    icon: string;
+  }[];
+  clouds: {
+    all: number;
+  };
+  wind: {
+    speed: number;
+    deg: number;
+    gust: number;
+  };
+}
+
+// format in which forecast for 5 days api returns the data for a given location
+interface WeatherData {
+  cod: string;
+  message: number;
+  cnt: number;
+  list: WeatherDetail[];
+  city: {
+    id: number;
+    name: string;
+    coord: {
+      lat: number;
+      lon: number;
+    };
+    country: string;
+    population: number;
+    timezone: number;
+    sunrise: number;
+    sunset: number;
+  };
+}
+
+// props of Linechart
+interface Chart {
+  dates:string,
+  temp:number
+}
+
+//getting icon based on current weather
+export function getIconUrl(code: string): string {
+  return `http://openweathermap.org/img/wn/${code}.png`;
+}
+
+export default function Home() {
+  const API_KEY = process.env.NEXT_PUBLIC_API_KEY;
+  const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL;
+  const [weatherData,setWeatherData] = useState<WeatherDetail>();
+  const [latitude,setLatitude] = useState(0);
+  const [longitude,setLongitude] = useState(0);
+  const [place,setPlace] = useState("Japan");
+  const [forecast,setForecast] = useState<WeatherData>();
+
+  const currentDate = new Date();
+  const year = currentDate.getFullYear();
+  const month = currentDate.getMonth() + 1;
+  const day = currentDate.getDate();
+  const todayDate= `${year}/${month}/${day}`;
+  
+  //page routing
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+  const handleOnChange = async (value:string) => {
+    setPlace(value);
+  };
+
+  useEffect(() => {
+    axios
+        .get(`https://api.openweathermap.org/geo/1.0/direct?q=${place}&limit=5&appid=${API_KEY}`) //getting coordinates based on place selected by user
```

Comment:
Might be better if you separate API calls to be another file.

Reply:
Moved it to the custom hook file.